app/src/page/firstRun/view.js

```diff
@@ -60,36 +84,92 @@ class FirstRunScreen extends React.PureComponent {
 
   handleContinuePressed = () => {
     const pageIndex = FirstRunScreen.pages.indexOf(this.state.currentPage);
-    if (pageIndex === (FirstRunScreen.pages.length - 1)) {
-      // Final page. Let the app know that first run experience is completed.
-      if (NativeModules.FirstRun) {
-        NativeModules.FirstRun.firstRunCompleted();
+    if (this.state.currentPage !== 'email-collect' &&
+        pageIndex === (FirstRunScreen.pages.length - 1)) {
+      this.closeFinalPage();
+    } else {
+      // TODO: Actions and page verification for specific pages
+      if (this.state.currentPage === 'email-collect') {
+        // handle email collect
+        this.handleEmailCollectPageContinue();
+      } else {
+        this.showNextPage();
       }
+    }
+  }
 
-      // Navigate to the splash screen
-      this.launchSplashScreen();
-    } else {
-      // TODO: Page transition animation?
-       this.state.currentPage = FirstRunScreen.pages[pageIndex + 1];
+  handleEmailCollectPageContinue() {
+    const { notify, addUserEmail } = this.props;
+    // validate the email
+    AsyncStorage.getItem(Constants.KEY_FIRST_RUN_EMAIL).then(email => {
+      if (!email || email.trim().length === 0 || email.indexOf('@') === -1) {
```

Comment:
Nice and simple 👍 
There’s an RFC conforming regex I know of, I’ll try to pull it up tomorrow